lib/charms/tls_certificates_interface/v3/tls_certificates.py

```diff
@@ -1885,3 +1925,46 @@ def _find_certificate_in_relation_data(self, csr: str) -> Optional[ProviderCerti
                 continue
             return provider_certificate
         return None
+
+    def _calculate_expiry_notification_time(
```

Comment:
This method should probably be outside of the class scope and tested on its own. We likely want to have a couple of tests pointing directly at it.

Reply:
Done